src/adapters/poi/specials/navigator_geolocalisation_poi.js

```diff
@@ -23,8 +23,14 @@ export default class NavigatorGeolocalisationPoi extends Poi {
     return window.__navigatorGeolocalisationPoi;
   }
 
-  async geolocate(options = { displayErrorModal: true }) {
-    await Geolocation.showGeolocationModalIfNeeded();
+  async geolocate(options = { displayErrorModal: true, displayDirectionModalIfNeeded: false }) {
+    let requestPosition = true;
+    if (options.displayDirectionModalIfNeeded) {
+      requestPosition = await Geolocation.showGeolocationModalIfNeeded();
+      if (!requestPosition) {
+        reject();
```

Comment:
This looks like a copy-paste: this causes a linting error and will fail at runtime as `reject` is undefined. Not sure what you needed here, maybe just `return`?

Reply:
@bbecquet indeed the reject() alone was an error.
Though, I tried many ways to "return early" in this function (return, reject, ...), but every time, when closing the modal with the "x" button, I end up having this error in my console, which seems unrelated but only happens when the return early is in the code 🤔 
![image](https://user-images.githubusercontent.com/1225909/96461197-622a6400-1224-11eb-928d-65af285783c8.png)

Reply:
The error isn't unrelated. It happens when you return early without setting a position on the `NavigatorGeolocalisationPoi`. As you didn't exit by rejecting, the calling code assumes everything went right and try to use the lat/lng property on this special POI object, but it doesn't have any…

Your problem can probably be solved by moving this block in the Promise below, so the method has only on exit point (the Promise with its two clear states), instead of two.

But I know this whole geolocation process is confusing, the previous code doesn't help. IMO it tries too much to be clever instead of using plain and simple isolated functions. The `showGeolocationModalIfNeeded` method is quite confusing to me, I'm never sure of how it should behave, hence my other comment.

Can we start by listing what `showGeolocationModalIfNeeded` should return in each case?

Reply:
Previously, `showGeolocationModalIfNeeded` displayed a popup if needed (hence, the use has not given permission yer, and has not been asked to do so), and was "blocking" execution. It was not returning anything IIUC.
So, in any case, any caller of `showGeolocationModalIfNeeded` would still need to ask for geolocation using `poi.geolocate(options)`, which could return the location, or throw an error if location access is forbidden.

Reply:
From now, we would like to know if user has accepted the popup, in which case we will try to call `geolocate()`, or do nothing if the popup has been dismissed.

So, `showGeolocationModalIfNeeded` does not really make sense now, we could name it as `isGeolocationModalAccepted` for instance. Its signature could be:

`const isGeolocationModalAccepted = function (): boolean`

Il would block until the user has clicked an option, and return :
- true when user clicked on accept button
- false when user clicked on close button

Reply:
I would just name such function `isGeolocationAccepted`, without referring to a modal at all, since it many cases there won't be one.
But it looks like in this PR we are mixing two things:
 - the decision to display a custom modal (depending on the platform, the user action, the feacture)
 - the decision of the user to accept the geolocation (no: closing the custom modal or refusing the native browser modal, yes: accepting the native browser modal) 
Do you think your proposition fills all the needs?

Reply:
Yes, but it's improvable. If you're ok with this, I will separate the modal logic and the native popup logic into 2 functions, and move the "if(requestPosition) back inside the promise.
After merge, we'll see if we want to improve that part more deeply (but I think it'll be enough)
OK?

Reply:
Great!